### ISSUE TITLE
Security issues reported by npm

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-ui#132]
At the moment there are three - very annoying - security issues reported by `npm`. They have been present, at least in my environment, for a few days now. And `npm audit fix` and `ncu -u` are not able to fix them.

```
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ jquery                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.4.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cytoscape-navigator                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cytoscape-navigator > jquery                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/796                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting (XSS)                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ jquery                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cytoscape-navigator                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cytoscape-navigator > jquery                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/328                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.12                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @vue/cli-plugin-e2e-cypress [dev]                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @vue/cli-plugin-e2e-cypress > cypress > lodash               │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1065                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 3 vulnerabilities (1 moderate, 2 high) in 38754 scanned packages
  3 vulnerabilities require manual review. See the full report for details.
```

This issue is to track the updates required to solve these issues, and so others are aware that it is a known issue.

**Comment by kinow** [comment]
These two:

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ jquery                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.4.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cytoscape-navigator                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cytoscape-navigator > jquery                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/796                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Cross-Site Scripting (XSS)                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ jquery                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ cytoscape-navigator                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ cytoscape-navigator > jquery                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/328                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

In Cytoscape, or more specifically in `cytoscape-navigator` are due to an older version of JQuery used. The author replied saying he doesn't have enough bandwidth to fix it now, but a user is working on a fix.

- https://github.com/cytoscape/cytoscape.js-navigator/issues/40

Once that's done and a new version has been released, `npm audit fix` and `ncu -u` should be able to fix it.

**Comment by kinow** [comment]
Now this one

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.12                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @vue/cli-plugin-e2e-cypress [dev]                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @vue/cli-plugin-e2e-cypress > cypress > lodash               │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1065                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

We also had the same issue coming from `lodash`. In this case, it happens in `@vue/cli-plugin-e2e-cypress`.

Their latest release is 3.4.0, which is [using](https://github.com/cypress-io/cypress/blob/v3.4.0/package.json) `"lodash": "4.17.11"` (advisory suggests using `>=4.17.12`).

But their [`develop` branch](https://github.com/cypress-io/cypress/blob/fa03fd9a63a357294280278105c13822c78bd58c/package.json#L133) is already on `4.17.14`, which satisfies the condition required.

So it is only pending a release of the vue plugin, so that `npm audit fix` and/or `ncu -u` can fix the issue.

**Comment by kinow** [comment]
Just checked, still two issues pending on Cytyoscape. Nothing that can be done right now.
